Manipulate distance information via CLI.  Fix #469

bbockelm opened the pull request:
This adds the ability to set, update, and get distance information between two RSEs via the `rucio-admin` CLI.

This only exposes the `ranking` and `distance` attributes: others will require usage of the REST API still.

Fixes #469.

bbockelm committed:
Manipulate distance information via CLI.

This adds the ability to set, update, and get distance information
between two RSEs via the `rucio-admin` CLI.

This only exposes the `ranking` and `distance` attributes: others
will require usage of the REST API still.

mlassnig reviewed (commented):
there isn't really a difference from the user point of view between set and update. i'd remove the update.

bbockelm commented:
@mlassnig - I thought the same thing.  Of course, internally, the map to two different APIs.

That would imply a loss of efficiency and atomicity as one would have to effectively first query whether the link already exists, then change the API call.  I think that's OK for the CLI ... thoughts?

I figure for the utmost efficiency and correctness, users should utilize the REST directly.

bari12 commented:
I think it is fine to have both, I am just wondering about naming it `set` and `update`. Update is fine I think, but setting something to me means updating it if it exists, or creating it if it doesn't. But here it just creates and does not update? So I wonder if it wouldn't be better to name it `add` (respective the CLI command `add-distance`) as well, similar to how the method is named in the python client.

vingar commented:
yes, `add` is more consistent (e.g., add rse, add account for HTTP POST operation).
For the update operation (HTTP PUT), `update` looks better for me here.

bbockelm committed:
clients: Change set-distance to add-distance.  #469

This changes the sub-command name from `set-distance` to `add-distance`
to better match the rest of the usage in the CLI.

bbockelm commented:
@vgaronne @mlassnig @bari12 - sorry for the delay (got sidetracked on other PRs).  The rename of the command is done, as requested.

Anything else I missed for this one?
